linera-chain/src/data_types.rs

```diff
@@ -612,16 +612,23 @@ pub struct ProposalContent {
 }
 
 impl BlockProposal {
-    pub fn new_initial(round: Round, block: ProposedBlock, secret: &AccountSecretKey) -> Self {
+    pub fn new_initial(
+        round: Round,
+        block: ProposedBlock,
+        signer: &(impl Signer + ?Sized),
+    ) -> Self {
+        let owner = block
+            .authenticated_signer
+            .expect("Block author is required for new block proposals");
```

Comment:
Since we're changing the signature of this function anyway, it would be nice to encode this in the type system by taking an `AccountOwner` as parameter.

Reply:
Account owner here is taken from the already-constructed block where this field is optional.

Reply:
This was incorrect - `block.authenticated_signer` was allowed to be `None`. I modified the constructors of `BlockProposal` to take the proposer's `AccountOwner` value explicitly. All tests have been updated.

Reply:
The None authenticated signer was allowed on purpose

Reply:
You could call that an "unauthenticated block"

Reply:
Yes, I learned that later on. This change is no longer part of the latest diff.